MassRegistrationOfSAPSystems/RegisterSIDs.ps1

```diff
@@ -67,38 +67,42 @@ foreach($line in $file)
     $MsiID = $line.MsiId
     $ManagedRgName = $line.ManagedResourceGroupName
     $ManagedRgStorageAccountName = $line.ManagedRgStorageAccountName
-    $TagVals = ($line.Tag -replace '\s','').Split(";")
-    $Keys=@()
-    $Values=@()
-
-    # Creating Tag Hash Table
-    foreach($TagVal in $TagVals)
+    $ManagedResourcesNetworkAccessType = $line.ManagedResourcesNetworkAccessType
+    $Tag = $line.Tag
+    
+    # Building the command to register the SID as a VIS resource
+    $command = "New-AzWorkloadsSapVirtualInstance -ResourceGroupName $ResourceGroup -Name $SID -Location $Location -Environment $Environment -SapProduct $Product -CentralServerVmId $CentralServerVmId $ArgManagedRgName -IdentityType 'UserAssigned' -UserAssignedIdentity @{'$MsiID'=@{}} -Tag @{$Tag}"
+    
+    # Checking if the optional parameters are provided in the input file and adding them to the command
+    if ($ManagedRgName -match '[a-zA-Z]' -or $ManagedRgName -match '[0-9]')
+    {
+        $command += " -ManagedResourceGroupName $ManagedRgName"
+    }
+    
+    if ($ManagedRgStorageAccountName -match '[a-zA-Z]' -or $ManagedRgStorageAccountName -match '[0-9]')
     {
-        $SubTag = $TagVal.Split("=")
-        $Keys += $SubTag[0]
-        $Values += $SubTag[1]
+        $command += " -ManagedResourceGroupStorageAccountName $ManagedRgStorageAccountName"
     }
-    $i=0
-    $Tag = @{}
-    foreach($Key in $Keys)
+
+    if ($ManagedResourcesNetworkAccessType -match '[a-zA-Z]' -or $ManagedResourcesNetworkAccessType -match '[0-9]')
```

Comment:
Can we match it to Public and Private as the only values supported if supplied.

Reply:
This is done